src/authentication/reset-password/index.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+
+import { Button, Input, Alert } from '@zero-tech/zui/components';
+
+import { bem, bemClassName } from '../../lib/bem';
+
+import './styles.scss';
+
+const c = bem('reset-password');
+const cn = bemClassName('reset-password');
+
+export interface Properties {
+  isLoading: boolean;
+  emailSubmitted: boolean;
+  errors: {
+    email?: string;
+    general?: string;
+  };
+
+  onSubmit: (data: { email: string }) => void;
+}
+
+interface State {
+  email: string;
+}
+
+export class ResetPassword extends React.Component<Properties, State> {
+  state = {
+    email: '',
+  };
+
+  handleResetPassword = (e) => {
+    e.preventDefault();
+    this.props.onSubmit({ email: this.state.email });
+  };
+
+  trackEmail = (value) => this.setState({ email: value });
+
+  get emailError() {
+    if (this.props.errors.email) {
+      return { variant: 'error', text: this.props.errors.email } as any;
+    }
+    return null;
+  }
+
+  get generalError() {
+    return this.props.errors.general;
+  }
+
+  get isSubmitDisabled() {
+    return this.props.isLoading || this.state.email.length === 0;
+  }
+
+  renderSuccessMessage() {
+    return (
+      <div {...cn('success-message')}>
+        An email containing a reset password link has been emailed to: {this.state.email}
+      </div>
+    );
+  }
+
+  renderForm() {
+    return (
+      <form {...cn('form')} onSubmit={this.handleResetPassword}>
+        <Input
+          {...cn('input')}
```

Comment:
I really don't like how we have to override the styles for all of these components. Are we sure these are all necessary? `input`, `submit-button`, `error`.... seems like something's wrong if we are constantly forceably overriding our UI lib styles.